resources.tf

```diff
@@ -150,42 +150,171 @@ resource "aws_iam_role_policy_attachment" "ec2_for_ssm_attachment" {
+
   statement {
     effect = "Allow"
 
     actions = [
-      "acm:ExportCertificate",
-      "cloudwatch:*",
+      "ec2:AuthorizeSecurityGroup*",
+      "ec2:CancelSpotInstanceRequests",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:Describe*",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyImageAttribute",
+      "ec2:ModifyInstanceAttribute",
+      "ec2:RequestSpotInstances",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RunInstances",
+      "ec2:TerminateInstances",
+      "ec2:DeleteVolume",
+      "ec2:DetachVolume",
+      "ec2:GetEbsDefaultKmsKeyId",
+      "iam:GetRole*",
+      "iam:List*",
+      "iam:PassRole",
+      "s3:CreateBucket",
+      "sqs:CreateQueue",
+      "sqs:Delete*",
+      "sqs:GetQueue*",
+      "sqs:PurgeQueue",
+      "sqs:ReceiveMessage",
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DescribeAlarms",
+      "cloudwatch:DeleteAlarms",
+      "application-autoscaling:RegisterScalableTarget",
+      "application-autoscaling:DeregisterScalableTarget",
+      "application-autoscaling:PutScalingPolicy",
+      "application-autoscaling:DeleteScalingPolicy",
+      "application-autoscaling:Describe*",
+      "elasticmapreduce:ListInstanceGroups",
+      "elasticmapreduce:ModifyInstanceGroups",
       "dynamodb:*",
-      "ec2:*",
-      "elasticmapreduce:*",
-      "kinesis:*",
-      "rds:Describe*",
-      "sdb:*",
-      "sns:*",
-      "sqs:*",
-      "glue:*",
-      "kms:*",
-      "iam:*",
-      "application-autoscaling:*",
-      "ssm:*",
-      "ssmmessages:*",
-      "ec2messages:*",
-      "cloudwatch:PutMetricData",
-      "ec2:DescribeInstanceStatus",
-      "ds:CreateComputer",
-      "ds:DescribeDirectories",
+      "glue:*Database",
+      "glue:*Table*",
+      "glue:*Partition",
+      "glue:*UserDefinedFunction",
+      "elasticmapreduce:Describe*",
+      "elasticmapreduce:List*",
+      "acm:ExportCertificate",
       "logs:*",
-      "s3:*",
-      "secretsmanager:*"
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:DescribeKey",
+      "kms:List*",
+      "kms:Create*",
+      "kms:Put*",
+      "secretsmanager:List*",
+      "secretsmanager:DescribeSecret",
+      "secretsmanager:Get*",
+      "secretsmanager:CreateSecret",
+      "secretsmanager:*Tag*",
     ]
 
     resources = [
       "*"
     ]
   }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      aws_s3_bucket.published.arn,
+      "${data.terraform_remote_state.common.outputs.config_bucket.arn}",
+      "arn:aws:s3:::${data.terraform_remote_state.ingest.outputs.s3_buckets.input_bucket}",
+      "arn:aws:s3:::${data.terraform_remote_state.security-tools.outputs.logstore_bucket.id}",
+    ]
+  }
+
+  statement {
```

Comment:
Can you split this into 2 please. The bit that handles the log bucket shouldn't ever be able to delete logs otherwise there's a potential for us to lose audit trails.

Reply:
Amended the code